specification/quantum/resource-manager/Microsoft.Quantum/preview/2019-11-04-preview/quantum.json

```diff
@@ -0,0 +1,910 @@
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Link to the next set of results. Not empty if Value contains incomplete list of Namespaces."
+        }
+      },
+      "description": "The response of a list Workspaces operation."
+    },
+    "ProviderListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ProviderDescription"
+          },
+          "description": "Result of a list Providers operation."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Link to the next set of results. Not empty if Value contains incomplete list of Namespaces."
+        }
+      },
+      "description": "The response of a list Providers operation."
+    },
+    "QuantumWorkspace": {
+      "type": "object",
+      "description": "The resource proxy definition object for quantum workspace.",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The unique id for this workspace.",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of this workspace.",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of this workspace.",
+          "readOnly": true
+        },
+        "location": {
+          "type": "string",
+          "description": "The location (region) of this workspace."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Gets or sets the properties. Define quantum space's specific properties.",
+          "$ref": "#/definitions/WorkspaceResourceProperties"
+        },
+        "tags": {
+          "type": "object",
+          "description": "The tags associated with this workspace."
+        },
+        "identity": {
+          "type": "object",
+          "description": "Managed Identity information",
+          "properties": {
+            "type": {
+              "type": "string",
+              "description": "Type of identity. Supported: SystemAssigned or None (case insensitive)"
+            },
+            "principalId": {
+              "type": "string",
+              "description": "objectId of the managed identity,"
+            },
+            "tenantId": {
+              "type": "string",
+              "description": "tenantId where the managed identity lives."
+            }
+          }
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "WorkspaceResourceProperties": {
+      "description": "Properties of a Workspace",
+      "properties": {
+        "providers": {
+          "type": "array",
+          "description": "List of Providers selected for this Workspace",
+          "items": {
+            "$ref": "#/definitions/Provider"
+          }
+        },
+        "usable": {
```

Comment:
this name seems to imply a bool - since you have a partial state too, consider using usabilityStatus or similar instead.

Reply:
I'm guessing this is a readonly property i.e. cannot be put/patch'ed - please declare it as such using readonly: true

Reply:
And make sure to mark any other properties that read-only as such as well.

---
In reply to: [546947868](https://github.com/Azure/azure-rest-api-specs/pull/11266#discussion_r546947868) [](ancestors = 546947868)

Reply:
Marked is as readonly, but renaming is a breaking change for existing customers. We'll take this at a new api version after the preview.

Reply:
You cannot make breaking changes between GA'd api-versions

Reply:
@JeffreyRichter We're going public preview, not GA yet. Also we don't have end customers directly calling APIs, but a number of clients - python, .NET, CLI and portal extension. Handling this via a new api-version will help us fix this with no downtime to customers.

Reply:
OK then, I understand